### PR TITLE
Updating ECC608 PAL port for OTA v3

### PR DESCRIPTION
**Pull request by divekarshubham**
Updating ECC608 PAL port for OTA v3

-----------

* Updated the PAL with the latest Error codes and structures
* Updated project files
* Updated cmake files to reflect components needed for OTA
* Added configuration files to support qualification tests

Checklist:
----------

- [ X ] I have tested my changes. No regression in existing tests. - Tested changes against E2E and Qualification tests.
- [ X ] My code is Linted. - Uncrustified code.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/core_mqtt_agent_config.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * will be.
+ */
+#define MQTT_AGENT_MAX_SUBSCRIPTION_FILTER_LENGTH    ( 100 )
+
+/**
+ * @brief Dimensions the buffer used to serialise and deserialise MQTT packets.
+ * @note Specified in bytes.  Must be large enough to hold the maximum
+ * anticipated MQTT payload.
+ */
+#define MQTT_AGENT_NETWORK_BUFFER_SIZE               ( 5000 )
```
[the remark this answers is not in the document]
Comment (divekarshubham) [reply vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/core_mqtt_agent_config.h:53]: Yes, block size is set to 1K.

File: vendors/microchip/boards/ecc608a_plus_winsim/aws_tests/config_files/aws_test_ota_config.h [hunk 2]
```diff
@@ -30,56 +30,43 @@
...
  * @brief Some boards OTA PAL layers will use the file names passed into it for the
  * image and the certificates because their non-volatile memory is abstracted by a
  * file system. Set this to 1 if that is the case for your device.
  */
-#define otatestpalUSE_FILE_SYSTEM     0
+#define otatestpalUSE_FILE_SYSTEM                          0
 
 /**
- * @brief 1 if prvPAL_CheckFileSignature() is implemented in aws_ota_pal.c.
+ * @brief 1 if otaPal_CheckFileSignature() is implemented in aws_ota_pal.c.
```
[the remark this answers is not in the document]
Comment (divekarshubham) [reply vendors/microchip/boards/ecc608a_plus_winsim/aws_tests/config_files/aws_test_ota_config.h]: Updated in https://github.com/aws/amazon-freertos/pull/3147/commits/21fe66897df4c4fb4611dbcf344b4259ae93d3af

File: vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/ota_config.h [hunk 3]
```diff
@@ -122,6 +141,17 @@
  */
 #define otaconfigMAX_NUM_OTA_DATA_BUFFERS       4U
```
[the remark this answers is not in the document]
Comment (divekarshubham) [reply vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/ota_config.h]: Thanks, updated in https://github.com/aws/amazon-freertos/pull/3147/commits/21fe66897df4c4fb4611dbcf344b4259ae93d3af

File: vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/ota_config.h [hunk 4]
```diff
@@ -56,11 +81,6 @@
  */
 #define otaconfigFILE_REQUEST_WAIT_MS           2500U
```
[the remark this answers is not in the document]
Comment (divekarshubham) [reply vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/config_files/ota_config.h]: Done, https://github.com/aws/amazon-freertos/pull/3147/commits/21fe66897df4c4fb4611dbcf344b4259ae93d3af

**Review by pvyawaha — APPROVED** [review]

**Pull request merged**